ci/docker/java-jni-manylinux-201x.dockerfile

```diff
@@ -18,23 +18,9 @@
 ARG base
 FROM ${base}
 
-# Install the libraries required by the Gandiva to run
-# Use enable llvm[enable-rtti] in the vcpkg.json to avoid link problems in Gandiva
-RUN vcpkg install \
-        --clean-after-build \
-        --x-install-root=${VCPKG_ROOT}/installed \
-        --x-manifest-root=/arrow/ci/vcpkg \
-        --x-feature=dev \
-        --x-feature=flight \
-        --x-feature=gcs \
-        --x-feature=json \
-        --x-feature=parquet \
-        --x-feature=gandiva \
-        --x-feature=s3
-
```

Comment:
This should remain in the image as people use this for local development too. 

But my understanding is that if you install the right triplet outside docker and mount it in the container for --x-install-root this should be a noop.